testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeDeltaLakeOss.java

```diff
@@ -0,0 +1,111 @@
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tests.product.launcher.docker.DockerFiles;
+import io.trino.tests.product.launcher.env.DockerContainer;
+import io.trino.tests.product.launcher.env.Environment;
+import io.trino.tests.product.launcher.env.EnvironmentConfig;
+import io.trino.tests.product.launcher.env.EnvironmentProvider;
+import io.trino.tests.product.launcher.env.common.Hadoop;
+import io.trino.tests.product.launcher.env.common.Minio;
+import io.trino.tests.product.launcher.env.common.Standard;
+import io.trino.tests.product.launcher.env.common.TestsEnvironment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+
+import javax.inject.Inject;
+
+import static io.trino.tests.product.launcher.docker.ContainerUtil.forSelectedPorts;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.HADOOP;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.TESTS;
+import static io.trino.tests.product.launcher.env.common.Minio.MINIO_CONTAINER_NAME;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_PRESTO_ETC;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TEMPTO_PROFILE_CONFIG;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+/**
+ * SinglenodeDeltaLakeOss test environment consists of:
+ * - Hive (used for metastore) (HDP 3.1)
+ * - Spark with open source implementation of Delta lake
+ * - Minio based S3 to store table data
+ */
+@TestsEnvironment
+public class EnvSinglenodeDeltaLakeOss
+        extends EnvironmentProvider
+{
+    private static final int SPARK_THRIFT_PORT = 10213;
+
+    private final DockerFiles dockerFiles;
+    private final PortBinder portBinder;
+    private final String hadoopImagesVersion;
+    private final DockerFiles.ResourceProvider configDir;
+
+    @Inject
+    public EnvSinglenodeDeltaLakeOss(
+            Standard standard,
+            Hadoop hadoop,
+            DockerFiles dockerFiles,
+            EnvironmentConfig config,
+            PortBinder portBinder,
+            Minio minio)
+    {
+        super(ImmutableList.of(standard, hadoop, minio));
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        this.portBinder = requireNonNull(portBinder, "portBinder is null");
+        this.hadoopImagesVersion = requireNonNull(config, "config is null").getHadoopImagesVersion();
+        this.configDir = dockerFiles.getDockerFilesHostDirectory("conf/environment/singlenode-delta-lake-oss");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        // Using hdp3.1 so we are using Hive metastore with version close to versions of  hive-*.jars Spark uses
+        builder.configureContainer(HADOOP, container -> {
```

Comment:
Would it make sense to open source an image containing only HMS instead of the entire Hadoop ecosystem?